packages/salesforce-adapter/src/filters/add_missing_ids.ts

```
@@ -84,10 +79,13 @@ const filter: FilterCreator = ({ client, config }) => ({
         metadataType,
       )
       log.debug(`Getting missing ids for the following types: ${Object.keys(groupedElements)}`)
-      await Promise.all(
+      const results = await Promise.allSettled(
         Object.entries(groupedElements)
           .map(([typeName, typeElements]) => addMissingIds(client, typeName, typeElements))
       )
+      if (results.some(r => r.status === 'rejected')) {
+        throw new Error('Failed to add missing ids')
```

this means we won't get the original error anywhere.
we should either throw the original error as the code used to do before this change or log the error(s) first.